src/lowlevel_api.jl

```diff
@@ -239,17 +239,17 @@ IteratorEltype(::Type{Object}) = Base.HasEltype()
 eltype(::Type{Object}) = Object
 isempty(::Object) = false
 function iterate(obj::Object)
-    state = vcat(obj.children, obj.memory_children, obj.io_children)
+    state = vcat(obj.memory_children, obj.children, obj.io_children)
     return obj, state
 end
 function iterate(::Object, state::Vector{Object})
     isempty(state) && return nothing
     # depth-first traversal
     # obj = shift!(state)
     obj, state = state[1], state[2:end]
+    prepend!(state, obj.io_children)
```

Comment:
Note that this order: `io_children -> children -> memory_children`, does not match the order above: `memory_children -> children -> io_children`. If this memory-children order is preferred then this should be:
```julia
prepend!(state, obj.memory_children)
prepend!(state, obj.children)
prepend!(state, obj.io_children)
```

Reply:
Actually, these are `prepend!` so it's "last prepended == first iterated"

Reply:
Hmm, maybe it should be `prepend!(state, reverse(obj.memory_children))` etc.

Reply:
Ah! this makes sense. I'm not thinking if we should switch to `push!` instead for code clarity? I inherited `prepend!` from @carstenbauer -- maybe you have thoughts.

Reply:
We need to use `prepend!` (rather than `push!`) here to keep things iterating in the proper order.  Upon further reflection, we do _not_ need to use `reverse` as I had been considering in an earlier comment.  I think adding a comment to these `prepend!` lines would be helpful to future readers so they don't have to "rediscover" the logic behind this.  I'll push another commit that adds a comment to these lines.